src/tdd/isograma.ts

```diff
@@ -1,9 +1,10 @@
-function isograma(string) {
+function isograma(s: string): boolean {
 
-        if (string !== '-') {
-          callback(string);
-        }}
+    for (let i = 0; i < s.length / 2; i++)
+    if (s[i] !=s[s.length - i - 1])
+      if (s > " ")
+    s.replaceAll("-"," ");
 
-
-
-export default isograma;
\ No newline at end of file
+    return false
+};
+export default isograma;
```

Comment:
1- Lembra de rodar o `yarn prettier`, essa formatação dificulta um pouco entender o que está sendo feito.
2- Não entendi muito bem a lógica, o for loop e o primeiro if fazem sentido, mas essa parte aqui não está alterando ou resolvendo nada:
```js
    if (s > " ")
        s.replaceAll("-", " ")
```
**Sobre o if:** existe uma parada que chama [tabela ASCII](https://www.rapidtables.com/code/text/ascii-table.html), que basicamente define um valor numérico pra cara caractere do alfabeto. Esse `if` está basicamente comparando o valor ASCII de `" "` com o da string recebida como input da função, e vai ser sempre verdadeiro (a não ser que a string `s` seja `""`).'

**Sobre o replaceAll:** que nem comentamos, o uso do `replaceAll` dessa maneira como está na função não faz nada na prática. A ideia é chamar o `replaceAll` antes do for loop para remover `" "` e `-`, e além disso o `replaceAll` não altera a string diretamente, então algo parecido com isso deve ser feito:
```js
let strSemCaracteresEspeciais = s.replaceAll("-", "")
strSemCaracteresEspeciais = strSemCaracteresEspeciais.replaceAll(" ", "")

for (let i = 0; i < strSemCaracteresEspeciais.length / 2; i++) {
...
```

Reply:
Entendi essa aqui!

Muito obrigado!